include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,267 @@
+  //! @}
+  //====================================================================================================================
+  struct relative_conditional_option
+  {
+    auto process(auto const& base, eve::relative_conditional_expr auto opt) const
+    {
+      auto new_opts = rbr::merge(options{condition_key = opt}, base);
+      return options<decltype(new_opts)>{new_opts};
+    }
+
+    /// Default settings of eve::relative_conditional is eve::ignore_none
+    EVE_FORCEINLINE constexpr auto default_to(auto const& base) const noexcept
+    {
+      auto new_opts = rbr::merge(base, options{condition_key = ignore_none});
+      return options<decltype(new_opts)>{new_opts};
+    }
+  };
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct conditional_option
+  //!   @brief Option specification for decoration via conditional value and expressions
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   eve::conditional_option is an option specification that can be used when defining a eve::callable type to make
+  //!   it supports decoration via `bool`, eve::logical, any eve::conditional_expr or eve::relative_conditional_expr.
+  //!
+  //!   @groupheader{Example}
+  //!
+  //!   @godbolt{doc/traits/callable_supports.cpp}
+  //! @}
+  //====================================================================================================================
+  struct conditional_option
+  {
+    auto process(auto const& base, rbr::concepts::exactly<condition_key> auto opt) const
+    {
+      auto new_opts = rbr::merge(base, rbr::settings{opt});
+      return options<decltype(new_opts)>{new_opts};
+    }
+
+    template<std::same_as<bool> O>
+    auto process(auto const& base, O opt) const
+    {
+      // Just delay the evaluation of the type by injecting some templates
+      using type = std::conditional_t<std::same_as<bool,O>,std::uint8_t,O>;
+      return process(base, condition_key = if_(logical<type>(opt)) );
+    }
+
+    auto process(auto const& base, eve::logical_value auto opt) const
+    {
+      return process(base, condition_key = if_(opt));
+    }
+
+    auto process(auto const& base, eve::conditional_expr auto opt) const
+    {
+      return process(base, condition_key = opt);
+    }
+
+    /// Default settings of eve::conditional is eve::ignore_none
+    EVE_FORCEINLINE constexpr auto default_to(auto const& base) const noexcept
+    {
+      auto new_opts = rbr::merge(base, options{condition_key = ignore_none});
+      return options<decltype(new_opts)>{new_opts};
+    }
+  };
+
+  /// Checks if the type associated to a given Keyword in a Option pack is equal to Type
+  template<auto Keyword, typename Opts, typename Type>
+  inline constexpr bool option_type_is =  std::same_as<Type, rbr::result::fetch_t<Keyword,Opts>>;
```

Comment:
concept? Why is this needed?

Reply:
It is used in some callable for doign thing like : 

```
// Are we dealing with a no-condition call ?
      if constexpr( option_type_is<condition_key, O, ignore_none_> )
      {
       // 
      }
```

I put it there but it def. should move to raberu

Reply:
So should be concept anyways